packages/workspace/src/browser/workspace-service.ts

```diff
@@ -318,11 +318,11 @@ export class WorkspaceService implements FrontendApplicationContribution {
     /**
      * Clears current workspace root.
      */
-    close(): void {
+    async close(): Promise<void> {
         this._workspace = undefined;
         this._roots.length = 0;
 
-        this.server.setMostRecentlyUsedWorkspace('');
+        await this.server.setMostRecentlyUsedWorkspace('');
```

Comment:
I'm looking at the implementation of `setMostRecentlyUserWorkspace`, here:

https://github.com/theia-ide/theia/blob/master/packages/workspace/src/node/default-workspace-server.ts#L105

Do you think the call to `writeToUserHome` would need to be awaited, to be sure it's completed before we try to reload?

Reply:
It doesn't seem necessary, when reloading we get the most recent path from memory (`this.root` in `DefaultWorkspaceServer`), not from the json file. So nevermind.